cl/citations/tests.py

```diff
@@ -1727,3 +1731,316 @@ def test_get_graph_component(self):
+        """Do we fail gracefully with invalid page numbers?"""
+        r = await self.async_client.post(
+            reverse("citation-lookup-list", kwargs={"version": "v3"}),
+            {
+                "reporter": "f2d",
+                "volume": "1",
+                "page": "asdf",
+            },
+        )
+
+        self.assertEqual(r.status_code, HTTPStatus.OK)
+        data = json.loads(r.content)
+        self.assertEqual(len(data), 1)
+
+        first_citation = data[0]
+        self.assertEqual(first_citation["citation"], "1 f2d asdf")
+        self.assertEqual(first_citation["status"], HTTPStatus.NOT_FOUND)
+        self.assertIn("Citation not found:", first_citation["error_message"])
+
+    async def test_can_match_citation_with_reporter_volume_page(self):
+        r = await self.async_client.post(
+            reverse("citation-lookup-list", kwargs={"version": "v3"}),
+            {"reporter": "f2d", "volume": "56", "page": "9"},
+        )
+
+        self.assertEqual(r.status_code, HTTPStatus.OK)
+        data = json.loads(r.content)
+        self.assertEqual(len(data), 1)
+
+        first_citation = data[0]
+        self.assertEqual(first_citation["citation"], "56 f2d 9")
+        self.assertEqual(first_citation["status"], HTTPStatus.OK)
+
+        clusters = first_citation["clusters"]
+        self.assertEqual(len(clusters), 1)
+        self.assertEqual(
+            clusters[0]["absolute_url"],
+            self.opinion_cluster_2.get_absolute_url(),
+        )
+
+        # Here opinion cluster 2 has the citation 56 F.2d 9, but the
+        # HTML with citations contains star pagination for pages 9 and 10.
+        # This tests if we can find opinion cluster 2 with page 9 and 10
+        r = await self.async_client.post(
+            reverse("citation-lookup-list", kwargs={"version": "v3"}),
+            {"reporter": "f2d", "volume": "56", "page": "10"},
+        )
+
+        self.assertEqual(r.status_code, HTTPStatus.OK)
+        data = json.loads(r.content)
+        self.assertEqual(len(data), 1)
+
+        first_citation = data[0]
+        self.assertEqual(first_citation["citation"], "56 f2d 10")
+        self.assertEqual(first_citation["status"], HTTPStatus.OK)
+
+        clusters = first_citation["clusters"]
+        self.assertEqual(len(clusters), 1)
+        self.assertEqual(
+            clusters[0]["absolute_url"],
+            self.opinion_cluster_2.get_absolute_url(),
+        )
+
+    async def test_can_handle_reporter_typos(self):
+        r = await self.async_client.post(
+            reverse("citation-lookup-list", kwargs={"version": "v3"}),
+            {"reporter": "F2d", "volume": "56", "page": "9"},
+        )
+        self.assertEqual(r.status_code, HTTPStatus.OK)
+        data = json.loads(r.content)
+        self.assertEqual(len(data), 1)
+
+        first_citation = data[0]
+        self.assertEqual(first_citation["citation"], "56 F2d 9")
```

Comment:
I'm wondering if our users would find it easier to debug if we returned our normalized citations in an additional key. Like, they sent F2d, so the `citation` key is `56 F2d 9`, but we fixed their input and got `56 F.2d 9`, which we put in `normalized_citation`? 

Beyond debugging, I bet people would appreciate this as a way of fixing slopping RAGs, so they give cleaner citations.

Easy to do?

Reply:
@mlissner I have a question regarding ambiguous citations. These citations typically have multiple potential reporters. Should the `normalized_citation` field display both potentially valid citations?

for example, the reporter `H.` is ambiguous and a request to the API endpoint might return clusters matching both `Handy` or `Haw`. Should the `normalized_citation` key include both possibilities separated by a comma?

The complete example looks like this:

```
POST: 
   {"reporter": "H.", "volume": "1", "page": "150"}
RESPONSE:
[
   {
      "citation": "1 H. 150",
      "status": 200,
      "normalized_citation": "1 Handy 150,  1 Haw 150"
      "clusters: {...2 clusters...}
    },
]
```

Reply:
That makes sense to me, but I'd use a list and change the key to `normalized_citations`, plural. That'd work, right?

Reply:
> I'd use a list and change the key to normalized_citations

got it! 

> That'd work, right?

Yes! we can do that.